src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1308,6 +1313,20 @@ internal NamedTypeSymbol GetFixedImplementationType(FieldSymbol field)
             return result;
         }
 
+        internal Cci.IFieldReference GetModuleVersionId(Cci.ITypeReference mvidType, CSharpSyntaxNode syntaxOpt, DiagnosticBag diagnostics)
+        {
+            PrivateImplementationDetails details = GetPrivateImplClass(syntaxOpt, diagnostics);
+            // Avoid creating a closure for the static constructor creator unless it is likely to be executed.
+            return details.HasStaticConstructor ?
+                details.GetOrAddModuleVersionId(mvidType, null) :
+                GetModuleVersionIdWithStaticConstructorCreation(mvidType, syntaxOpt, diagnostics);
+        }
+
+        private Cci.IFieldReference GetModuleVersionIdWithStaticConstructorCreation(Cci.ITypeReference mvidType, CSharpSyntaxNode syntaxOpt, DiagnosticBag diagnostics) =>
+            PrivateImplClass.GetOrAddModuleVersionId(
+                mvidType,
+                () => new SynthesizedPrivateImplementationDetailsStaticConstructor(SourceModule, PrivateImplClass, GetUntranslatedSpecialType(SpecialType.System_Void, syntaxOpt, diagnostics)));
```

Comment:
It doesn't feel right that we are bundling creation of the static constructor with allocation of ModuleVersionId in this manner. In the future, there could be other scenarios when we might need to add static constructor unrelated to the field. Consider adding a separate method EnsurePrivateImplementationDetailsStaticConstructor in this class. GetModuleVersionId should simply call it, before returning the field.